beacon-chain/state/state-native/setters_misc.go

```diff
@@ -110,7 +109,7 @@ func (b *BeaconState) SetHistoricalRoots(val [][]byte) error {
 
 // SaveValidatorIndices save validator indices of beacon chain to cache
 func (b *BeaconState) SaveValidatorIndices() {
-	if !features.Get().EIP6110ValidatorIndexCache {
+	if b != nil && b.Version() < version.Electra {
```

Comment:
If `b` is nil, the function will continue and panic at the next line. It should be `b == nil ||` instead. Either way I think ni-checking receivers is too paranoid.